server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -248,7 +248,17 @@ private synchronized void updateSkipUnavailable(String clusterAlias, Boolean ski
     @Override
     protected void updateRemoteCluster(String clusterAlias, Settings settings) {
         CountDownLatch latch = new CountDownLatch(1);
-        updateRemoteCluster(clusterAlias, settings, ActionListener.wrap(latch::countDown));
+        updateRemoteCluster(clusterAlias, settings, ActionListener.runAfter(new ActionListener<Void>() {
+            @Override
+            public void onResponse(Void o) {
+                logger.debug("connected to new remote cluster {}", clusterAlias);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.debug("initial connection to new remote cluster {} failed", clusterAlias);
```

Comment:
I think "initial" is a bit confusing here, since it could be an update of a remote cluster that was previously connected and it could now reconnect?
```suggestion
                logger.debug("connection to new remote cluster {} failed", clusterAlias);
```